src/theory/arrays/theory_arrays.cpp

```diff
@@ -831,16 +832,15 @@ Node TheoryArrays::explain(TNode literal) {
   return explanation;
 }
 
-Node TheoryArrays::explain(TNode literal, eq::EqProof *proof)
-{
+Node TheoryArrays::explain(TNode literal, std::shared_ptr<eq::EqProof> proof) {
```

Comment:
It seems like `std::shared_ptr<EqProof> eqp` should be a `std::weak_ptr` since `explain` is not taking (shared) ownership.